pkg/image/association.go

```diff
@@ -204,42 +220,35 @@ func (as AssociationSet) validate() error {
 		}
 		for _, a := range assocs {
 
-			if s, ok := imageTypeStrings[a.Type]; ok && s != "" {
-				continue
-			}
-			switch a.Type {
-			case TypeInvalid:
-				// TypeInvalid is the default value for the concrete type, which means the field was not set.
-				errs = append(errs, fmt.Errorf("image %q: must set image type", a.Name))
-			default:
-				errs = append(errs, fmt.Errorf("image %q: unknown image type %v", a.Name, a.Type))
-			}
-
-			if len(a.ManifestDigests) != 0 && len(a.LayerDigests) != 0 {
-				errs = append(errs, fmt.Errorf("image %q: child descriptors cannot contain both manifests and image layers", a.Name))
-			}
-			if len(a.ManifestDigests) == 0 && len(a.LayerDigests) == 0 {
-				errs = append(errs, fmt.Errorf("image %q: child descriptors must contain at least one manifest or image layer", a.Name))
-			}
-
-			if a.ID == "" {
-				errs = append(errs, fmt.Errorf("image %q: tag or ID must be set", a.Name))
+			if err := a.validate(); err != nil {
+				errs = append(errs, err)
 			}
 		}
 	}
 	return utilerrors.NewAggregate(errs)
 }
 
-func GetImageFromBlob(as AssociationSet, digest string) string {
-	for imageName, assocs := range as {
-		for _, assoc := range assocs {
-			for _, dgst := range assoc.LayerDigests {
-				if dgst == digest {
-					return imageName
-				}
+func (a Association) validate() error {
 
-			}
-		}
+	if len(a.ManifestDigests) != 0 && len(a.LayerDigests) != 0 {
+		return fmt.Errorf("image %q: child descriptors cannot contain both manifests and image layers", a.Name)
+	}
+	if len(a.ManifestDigests) == 0 && len(a.LayerDigests) == 0 {
+		return fmt.Errorf("image %q: child descriptors must contain at least one manifest or image layer", a.Name)
+	}
+
+	if a.ID == "" && a.TagSymlink == "" {
+		return fmt.Errorf("image %q: tag or ID must be set", a.Name)
+	}
+
+	if s, ok := imageTypeStrings[a.Type]; ok && s != "" {
+		return nil
+	}
+	switch a.Type {
```

Comment:
Do you think it would add some clarity if we change the name of `TypeInvalid`.? An unknown image is meant to represent an `ImageType` that is not unset but does not have a corresponding string representation. Invalid (which may be misleading) is for ImageTypes that are unset or left as the default.

Reply:
You actually confused me more. I don't think this is something worth blocking this PR. It is not a big deal.